internal/eval/eval.go

```diff
@@ -3484,3 +3484,38 @@ func evalJSONNUMINCRBY(args []string, store *dstore.Store) []byte {
 	obj.Value = jsonData
 	return clientio.Encode(resultString, false)
 }
+func evalSETEX(args []string, store *dstore.Store) []byte {
+	if len(args) != 3 {
+		return diceerrors.NewErrArity("SETEX")
+	}
+
+	var key, value string
```

Comment:
@AshikBN Thanks for contributing.
Can we make use of existing [evalSET](https://github.com/DiceDB/dice/blob/df66d8cc45e5af8214f264536333c14cbde76ba5/internal/eval/eval.go#L116) function ?  

since
```
SETEX key seconds value
```
 is equivalent to
 
 ```
 SET key value EX seconds
``` 
Let me know if i am missing something.

Reply:
if your asking, why do we need setex command as we can use set also

from jyotinder :
We still need it because redis exporter for Prometheus makes use of this command.
Even though we can change the logic directly in the exporter to use the new preferred way, but it's not the highest priority right now. We should first focus on getting things working as they are.

if you are asking,why I wrote a separate function to implement setex rather than using evalSet itself
-in setex,expiry is mandatory and it will be in seconds and it does not accept any other commands
-if I had make use of evalset function,then I had to make change in that function to check if the command is set or setex ,then based on that I had make sure expiry is mandatory and consider that as seconds

then I had to to add conditions to call evalset in case of setex command.
this would have been an antipattern as we kept different eval functions for different command.
just wanted to keep the flow clean.

happy to make changes if you have any suggestions.

Reply:
@AshikBN
> if you are asking,why I wrote a separate function to implement setex rather than using evalSet itself

 yes i was talking about this.

i was thinking something like this

```
func evalSETEX(args []string, store *dstore.Store) []byte {
	if len(args) != 3 {
		return diceerrors.NewErrArity("SETEX")
	}
	var key, value string
	key, value = args[0], args[2]

	exDuration, err := strconv.ParseInt(args[1], 10, 64)
	if err != nil {
		return diceerrors.NewErrWithMessage(diceerrors.IntOrOutOfRangeErr)
	}
	if exDuration <= 0 || exDuration >= maxExDuration {
		return diceerrors.NewErrExpireTime("SETEX") 
	}

	newArgs := []string{key, value, Ex, strconv.FormatInt(exDuration, 10)}

	return evalSET(newArgs, store)
}

```
what do you think?

Reply:
yeah can be done.
but making another function call that does the same thing of updating the hash table. looks like an extra function call.
what do you suggest..? @kushal0511-not

Reply:
I am okay with your approach also , i just thought of redundant code.
May be @lucifercr07 or @JyotinderSingh can give opinion on this.

Reply:
Let's follow what Kushal is suggesting here. That is the standard way of handling deprecated implementations.
Deprecated APIs should be implemented using newer supported APIs.

Reply:
sure thank you